components/br-stream/src/event_loader.rs

```diff
@@ -190,20 +193,18 @@ where
         let mut event_loader = EventLoader::load_from(snap, start_ts, TimeStamp::max(), region)?;
         let mut stats = StatisticsSummary::default();
         loop {
-            let mut events = Vec::with_capacity(2048);
+            let mut events = ApplyEvents::with_capacity(1024, region.id);
             let stat = event_loader.scan_batch(1024, &mut events)?;
-            if events.is_empty() {
+            if events.len() == 0 {
```

Comment:
wouldn't clippy have warned that you have implemented `len()` but not `is_empty()`? 🤔

Reply:
Nope...
<img width="834" alt="image" src="https://user-images.githubusercontent.com/36239017/154018709-3adb13f6-b11d-4e40-9886-7a38335fe649.png">

Reply:
okay 🤔